lua/plugins/configs/lspconfig.lua

```diff
@@ -32,8 +32,8 @@ local function on_attach(_, bufnr)
    buf_set_keymap("n", "<space>ca", "<cmd>lua vim.lsp.buf.code_action()<CR>", opts)
    buf_set_keymap("n", "gr", "<cmd>lua vim.lsp.buf.references()<CR>", opts)
    buf_set_keymap("n", "<space>e", "<cmd>lua vim.lsp.diagnostic.show_line_diagnostics()<CR>", opts)
-   buf_set_keymap("n", "[d", "<cmd>lua vim.lsp.diagnostic.goto_prev()<CR>", opts)
-   buf_set_keymap("n", "]d", "<cmd>lua vim.lsp.diagnostic.goto_next()<CR>", opts)
+   buf_set_keymap("n", "[d", "<cmd>lua vim.lsp.diagnostic.goto_prev()<CR>zz", opts)
+   buf_set_keymap("n", "]d", "<cmd>lua vim.lsp.diagnostic.goto_next()<CR>zz", opts)
```

Comment:
@G-Rowell I think most of the part in this PR ( except for this lsp keymap) is user specific , can be added after the custom hooks PR is merged right?

Reply:
I don't think these are user specific. They should be builiting. Cause these are diagnostic prevs

Reply:
Agree with ashin, these could do well in base NvChad

Reply:
> I don't think these are user specific. They should be builitin. Cause these are diagnostic prevs

No i dont mean them , I meant these : 

![image](https://user-images.githubusercontent.com/59060246/133428857-440a102c-03ff-438d-8b71-5b7741187e49.png)

Reply:
bruh these are improvements in the builiting vim bindings not user bindings.

Reply:
Hmmmm, 

I do think that some of these mappings should be checked/tested.
Also they hard code their mapping key, rather than grab from `default_config` / `chadrc`.

I think we've also tried, and not been keen on `highlight on yank` being included by default?


Hmmm, not sure

Reply:
> bruh these are improvements in the builiting vim bindings not user bindings.

bruh the user can add this himself once the custom hooks PR gets merged

Reply:
> Hmmmm,
> 
> I do think that some of these mappings should be checked/tested.
> Also they hard code their mapping key, rather than grab from `default_config` / `chadrc`.
> 
> I think we've also tried, and not been keen on `highlight on yank` being included by default?
> 
> Hmmm, not sure

check primes video on must add keymappings for vim. https://www.youtube.com/watch?v=hSHATqh8svM

Reply:
> Hmmmm,
> 
> I do think that some of these mappings should be checked/tested.
> Also they hard code their mapping key, rather than grab from `default_config` / `chadrc`.
> 
> I think we've also tried, and not been keen on `highlight on yank` being included by default?
> 
> Hmmm, not sure

Maybe highlight on yank may not be needed and we could add it as an option for the user or smtng

Reply:
Dont think so , once the custom hooks PR is merged , that can be just added as an autocmd

Reply:
Have u seen primes video. These are must add mappings. The highlightonyank might not be needed but others are

Reply:
i dont watch prime's vids :angry: :smile:

Reply:
I watch em.

Reply:
Would agree with Ashin here,
though this PR does need some small housekeeping regardless

Reply:
> bruh the user can add this himself once the custom hooks PR gets merged

I agree with siduck here, why not make a dedicated section for `recommended default remaps` in wiki or something rather than hardcoded the default remaps like this? I think it would be easier for new user to add new mapping rather than remove existing mapping.

Reply:
While I do think that's a fair point,

NvChad is already pretty opinionated out of the box, that's what it's designed to be.

We want to provide great & reasonable defaults to everyone, so the question becomes:

> Is this mapping a good & sensible default, is it non-language specific & something that a majority of users would beneift from?

I think the answer is yes here imo.

Though I completely agree that we should always aim to have some way of turning off such things

Reply:
Adding an option to turn off the default would be ideal. I just don't want the tragedy of vim-sneak happen here too. Imagine you need to make a mapping just to disable the default mapping like this `nnoremap <F99> <plug>mapping1<plug>mapping2` (that's actually what justinmk suggested). Ugh, that's horrible.

Reply:
I agree with @bruhtus , nvchad needs to have less config as possible since its meant to be used as a base config and the user adds his/ her stuff on the top of it.

Reply:
Oh lord yeah, using a mapping to disable a plugin is not ideal.

We already have `default_config` / `chadrc` options to disable some in-built features.

[Like here](https://github.com/NvChad/NvChad/blob/main/lua/default_config.lua#L54] (though this should be put in a separate table to normal NeoVim options)